addons/sale_stock/sale_stock.py

```diff
@@ -309,25 +328,9 @@ def product_id_change_with_wh(self, cr, uid, ids, pricelist, product, qty=0,
         warning_msgs = res_packing.get('warning') and res_packing['warning']['message'] or ''
 
         if product_obj.type == 'product':
+            warehouse = warehouse_obj.browse(cr, uid, warehouse_id, context=context) if warehouse_id else False
```

Comment:
well no need to create browse record preventively as anyway it's only used in your _check_routing method